crates/compiler/builtins/roc/Hash.roc

```diff
@@ -66,3 +71,13 @@ Hasher has
     ## Completes the hasher, extracting a hash value from its
     ## accumulated hash state.
     complete : a -> U64 | a has Hasher
+
+## Adds a string into a [Hasher] by hashing its UTF-8 bytes.
+hashStrBytes = \hasher, s ->
+    Str.walkUtf8WithIndex s hasher \accumHasher, byte, _ ->
```

Comment:
Either way is fine with me, but couldn't the `Hasher` implementer decide to only process chunks once the defined chunk size is hit? You don't necessarily need to process one byte at a time, since you know you only have to spit out a hash once `complete` is called.

Reply:
I guess that is fair, but it comes at a performance cost. Every hasher must now include a larger state of at least their block size. Also, you now are essentially always forcing a memcpy in small blocks. I guess we can test this, but I would guess it would hurt performance too much. Let me try to write it up and see the actual impact. So I guess for now, leave this as is and we can circle back to it if the extra state management is too costly.

Reply:
Cool, that sounds good. Let's leave as-is for now so that you can make the judgement call later as to whether we need to change this. I imagine the implementation of WyHash and then some real applications will show what this actually looks like.